### DISCUSSION TITLE
Using 'browser.call' from within a constructor in async context

### DISCUSSION THREAD

**Discussion by tzurp** [webdriverio/webdriverio#7591] [Q&A] [answered]
In the wdio documentations for the command 'call' it is noted that this command is not so useful after deprecating the synchronous context in wdio.
Our project has some calls to methods from the class constructor that after migration to async mode became impossible because constructors cannot be asynchronous in JS. Thus, you can't call async method from within a constructor. Would it be 'proper' to use browser.call to call async methods from the constructor using async context?

**Comment by christian-bromann** [comment, marked as answer]
> Would it be 'proper' to use browser.call to call async methods from the constructor using async context?

No. The `call` command returns a promise you need to resolve which you can't in a constructor. I recommend create something like an `init` method to initiate your PO. Not sure what your classes do but I hope they don't contain any element commands in their constructor.
